packages/aws-cdk/lib/context-providers/index.ts

```diff
@@ -98,7 +97,6 @@ export async function provideContextValues(
       }
     }
 
-    ioHelper = ioHelper ?? CliIoHost.instance().asIoHelper();
```

Comment:
this was left over. `ioHelper` is a required argument, so the second part never happens anyway